mm2src/mm2_main/src/lp_swap/maker_swap.rs

```diff
@@ -1375,13 +1371,13 @@ impl MakerSwap {
         Ok((swap, command))
     }
 
-    pub async fn recover_funds(&self) -> Result<RecoveredSwap, String> {
-        async fn try_spend_taker_payment(selfi: &MakerSwap, secret_hash: &[u8]) -> Result<TransactionEnum, String> {
+    pub async fn recover_funds(&self) -> Result<RecoveredSwap, RecoverSwapError> {
+        async fn try_spend_taker_payment(selfi: &MakerSwap) -> Result<TransactionEnum, RecoverSwapError> {
```

Comment:
By the way, why not move `try_spend_taker_payment` function out of the `recover_funds` function? 

Right now, there is a large function nested inside another large function, and both only require `&self`.

Reply:
i think this scopes the helper function better since it's not used elsewhere. we can consider putting helpers on the `impl Struct` scope if that `impl` scope doesn't have too many methods already, but this one has tons of methods so scoping the helper inside `recover_funds` would make more sense.